src/mappings/mappingUniquesHandlers.ts

```diff
@@ -1,28 +1,150 @@
+  const collectionId = event.event.data[0];
+  const itemId = event.event.data[1];
+  const data = event.event.data[2];
+  const blockNumber = event.block.block.header.number.toNumber().toString();
+
+  const collection = await ensureCollection(collectionId.toString(), blockNumber);
+  const item = await ensureItem(collectionId.toString(), itemId.toString(), blockNumber);
+  item.metadataCid = data.toHuman().toString();
+  item.collectionId = collection.id;
+  await collection.save()
+  return item.save();
+}
+
+export const handleUniquesCollectionMetadataSetEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "uniqueCollectionMetadataSetEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+  const data = event.event.data[1];
+  const blockNumber = event.block.block.header.number.toNumber().toString();
+
+  const collection = await ensureCollection(collectionId.toString(), blockNumber);
+  collection.metadataCid = data.toHuman().toString();
+
+  return collection.save();
+}
+
+export const handleUniquesDestroyedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesDestroyedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+  const blockNumber = event.block.block.header.number.toNumber().toString();
+
+  const collection = await ensureCollection(collectionId.toString(), blockNumber);
+  collection.isDestroyed = true;
+  return collection.save()
+}
+
+export const handleUniquesBurnedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesBurnedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const itemId = event.event.data[1];
+  const collectionId = event.event.data[0];
+
+  const item = await ensureItem(collectionId.toString(), itemId.toString(), event.block.block.header.number.toNumber().toString());
+  item.isBurned = true;
+  return item.save();
+}
+
+export const handleUniquesIssuedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesIssuedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+  const itemId = event.event.data[1];
+  const owner = event.event.data[2];
+  const blockNumber = event.block.block.header.number.toNumber().toString();
+
+  const collection = await ensureCollection(collectionId.toString(), blockNumber);
+  const item = await ensureItem(collectionId.toString(), itemId.toString(), blockNumber);
+  item.owner = owner.toString();
+  item.collectionId = collection.id;
+  item.isBurned = false;
+  await collection.save()
+  return item.save();
+}
+
+export const handleUniquesCreatedEvent = async (
+  event: SubstrateEvent
+) => {
+  logger.debug(
+    "handleUniquesCreatedEvent added: " + JSON.stringify(event.toHuman())
+  );
+  const collectionId = event.event.data[0];
+  const blockNumber = event.block.block.header.number.toNumber().toString();
```

Comment:
@Douglasacost can you please check if we can improve on the uniqueness of the id by using `event.idx`. Events in substrate are identified uniquely by the mix of a block number and the event number. Together they would look like `1500000-3` as an example. Please check if `idx` is the mixed result or you'd need to add that as a suffix to the block number.